hs_core/views/__init__.py

```diff
@@ -236,21 +236,22 @@ def delete_metadata_element(request, shortkey, element_name, element_id, *args,
 
 
 def delete_file(request, shortkey, f, *args, **kwargs):
-    res, _, user = authorize(request, shortkey, edit=True, full=True, superuser=True)
+    res, _, user = authorize(request, shortkey, needed_permission=Action_To_Authorize.EDIT_RESOURCE)
     hydroshare.delete_resource_file(shortkey, f, user)
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
 
 def delete_resource(request, shortkey, *args, **kwargs):
-    res, _, _ = authorize(request, shortkey, edit=True, full=True, superuser=True)
+    res, _, _ = authorize(request, shortkey, needed_permission=Action_To_Authorize.DELETE_RESOURCE)
 
     res.delete()
     return HttpResponseRedirect('/my-resources/')
 
 
 def publish(request, shortkey, *args, **kwargs):
-    res, _, _ = authorize(request, shortkey, edit=True, full=True, superuser=True)
+    # only resource owners are allowed to change resource flags (e.g published)
+    res, _, _ = authorize(request, shortkey, needed_permission=Action_To_Authorize.SET_RESOURCE_FLAG)
```

Comment:
@pkdash On second thought, perhaps SET_RESOURCE_FLAG action is sufficient for publishing a resource since only owners are allowed to set resource flags.